src/leetCodeManager.ts

```diff
@@ -34,10 +34,44 @@ class LeetCodeManager extends EventEmitter {
         }
     }
 
-    public async signIn(isByCookie: boolean = false): Promise<void> {
-        const loginArg: string = "-l";
-        const cookieArg: string = "-c";
-        const commandArg: string = isByCookie ? cookieArg : loginArg;
+    public async signIn(): Promise<void> {
+        const picks: Array<IQuickItemEx<string>> = [];
+        picks.push(
+            {
+                label: "LeetCode Account",
+                description: "",
+                detail: "Use LeetCode account to login",
+                value: "LeetCode",
+            },
+            {
+                label: "LeetCode Cookie",
+                description: "",
+                detail: "Use LeetCode cookie that copy from browser to login",
+                value: "Cookie",
+            },
+            {
+                label: "Third-Party: GitHub",
+                description: "",
+                detail: "Use third party GitHub account to login",
+                value: "GitHub",
+            },
+            {
+                label: "Third-Party: LinkedIn",
+                description: "",
+                detail: "Use third party LinkedIn account to login",
+                value: "LinkedIn",
+            },
+        );
+        const choice: IQuickItemEx<string> | undefined = await vscode.window.showQuickPick(picks);
+        if (!choice) {
+            return;
+        }
+        const loginMethod: string = choice.value;
+        const commandArg: string | undefined = loginCommand.get(loginMethod);
+        if (!commandArg) {
+            throw new Error(`The login method "${loginMethod}" is not supported.`);
+        }
+        const isByCookie: boolean = loginMethod === "Cookie";
         const inMessage: string = isByCookie ? "sign in by cookie" : "sign in";
```

Comment:
Is this variable `inMessage ` used anywhere?

Reply:
Yes, it is for the difference of  cookie login or other login message.
e.g.
> promptForOpenOutputChannel(`Failed to ${inMessage}. Please open the output channel for details`